File: dotnet/src/fit/fitnesse/fixtures/TableFixture.cs
using fit;
// Modified or written by Object Mentor, Inc. for inclusion with FitNesse.
// Copyright (c) 2002 Cunningham & Cunningham, Inc.
// Released under the terms of the GNU General Public License version 2 or later.
namespace fitnesse.fixtures
{
	public abstract class TableFixture : Fixture
	{
		private Parse rows;
		
		protected abstract void DoStaticTable(int rows);

		public override void DoRows(Parse rows)
		{
			this.rows = rows;
			DoStaticTable(rows.Size);
		}

		protected Parse GetCell(int row, int column)
		{
			return rows.At(row, column);
		}

		protected string GetString(int row, int column)
		{
			Parse cell = GetCell(row, column);
			if (cell == null)
			{
				return null;
			}
			return cell.Text;
		}

		protected void Right(int row, int column)
		{
			Parse cell = rows.At(row, column);
			Right(cell);
		}

		protected void Wrong(int row, int column)
		{
			Parse cell = rows.At(row, column);
			Wrong(cell);
		}

		protected void Wrong(int row, int column, string actual)
		{
			Parse cell = rows.At(row, column);
			Wrong(cell, actual);
		}

		protected void Ignore(int row, int column)
		{
			Parse cell = rows.At(row, column);
			Ignore(cell);
		}

		protected int GetInt(int row, int column)
		{
			TypeAdapter adapter = new TypeAdapter(typeof(int));
			return (int) adapter.Parse(GetString(row, column));
		}

		protected bool Blank(int row, int column)
		{
			Parse cell = rows.At(row, column);
			return "".Equals(cell.Text.Trim());
		}
	}
}

File: trunk/src/fitnesse/fixtures/TableFixture.java
// Copyright (C) 2003,2004,2005 by Object Mentor, Inc. All rights reserved.
// Released under the terms of the GNU General Public License version 2 or later.
package fitnesse.fixtures;

import fit.*;
import fit.exception.FitFailureException;

public abstract class TableFixture extends Fixture
{
	protected Parse firstRow;

	public void doRows(Parse rows)
	{
		firstRow = rows;
		if(rows == null)
			throw new FitFailureException("There are no rows in this table");
		doStaticTable(rows.size());
	}

	protected abstract void doStaticTable(int rows);

	protected Parse getCell(int row, int column)
	{
		return firstRow.at(row, column);
	}

	protected String getText(int row, int column)
	{
		return getCell(row, column).text();
	}

	protected boolean blank(int row, int column)
	{
		return getText(row, column).equals("");
	}

	protected void wrong(int row, int column)
	{
		wrong(getCell(row, column));
	}

	protected void right(int row, int column)
	{
		right(getCell(row, column));
	}

	protected void wrong(int row, int column, String actual)
	{
		wrong(getCell(row, column), actual);
	}

	protected void ignore(int row, int column)
	{
		ignore(getCell(row, column));
	}

	protected int getInt(int row, int column)
	{
		int i = 0;
		String text = getText(row, column);
		if(text.equals(""))
		{
			ignore(row, column);
			return 0;
		}
		try
		{
			i = Integer.parseInt(text);
		}
		catch(NumberFormatException e)
		{
			wrong(row, column);
		}
		return i;
	}
}
